src/commands/retrieve.ts

```diff
@@ -170,14 +176,30 @@ export default function retrieve(resource?: vscode.Uri | ToolingTypes) {
             if(totalTypes > 10000) {
                 reject({ message: 'Cannot retrieve more than 10,000 files at a time. Please select "Choose Types..." from the retrieve menu and try to download without Reports selected first.'});
             }
-            var theStream = vscode.window.forceCode.conn.metadata.retrieve({
-                unpackaged: retrieveTypes,
-                apiVersion: vscode.window.forceCode.config.apiVersion || vscode.workspace.getConfiguration('force')['defaultApiVersion'],
-            });
-            theStream.on('error', (error) => {
-                reject(error || { message: 'Cannot retrieve more than 10,000 files at a time. Please select "Choose Types..." from the retrieve menu and try to download without Reports selected first.' });
-            });
-            resolve(theStream.stream());
+            var theStream;
+            try {
+                theStream = vscode.window.forceCode.conn.metadata.retrieve({
+                    unpackaged: retrieveTypes,
+                    apiVersion: vscode.window.forceCode.config.apiVersion || constants.API_VERSION,
+                });
+            } catch(e) {
+                reject(e);
+            }
+            if(theStream) {
+                theStream
+                    .on('progress', (result) => {
+                        CommandViewService.getInstance().updateLabel(task, `Retrieving package (${result.id} - ${result.state})`);
```

Comment:
You can use commandViewService instead of CommandViewService.getInstance(), it's previously declared as a constant